### PR TITLE
Changed API routes and functions to take `username` of user along with `id`

### LINKED ISSUE #94: Change API routes such that they take `username` instead of `id`
Currently our APIs like `pullrequests/:id` and `users/:id` take `id` as a path param. 
We need to change it so that it takes `username` as path param and proceed in the exact same way.

### PR DESCRIPTION
**Pull request by swarajpure**
Changed following API routes:
- `pullrequests/:id`
- `users/:id`

Also changed corresponding functions to accomodate and work with `username` of user ~instead of~ along with `id` of user

Fixes #94 

Please review. Thank you!

### CODE REVIEW

File: middlewares/authenticate.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ module.exports = async (req, res, next) => {
     const decoded = authService.verifyAuthToken(token)
 
     // add user data to `req.userData` for further use
-    const userData = await users.fetchUser(decoded.userId)
+    const userData = await users.fetchUser(null, decoded.userId)
```
Comment (whyDontI) [inline middlewares/authenticate.js:37]: **Suggestion: **
Passing `null` as an argument does not feel good.
Is it a better idea to make the argument as an Object which will contain `userId` and `userName`

The function call will look like this
```
const userData = await users.fetchUser({
    userId: 21389374,
    userName: 'Swarajpure'
})
```

Opinions: @ankurnarkhede @swarajpure

Comment (swarajpure) [reply middlewares/authenticate.js]: Awesome idea @whyDontI !
Had not at all thought of it.
Pushed commit with this change. Please review.

Thank you so much! 😄

File: models/users.js [hunk 2]
```diff
@@ -79,20 +79,31 @@ const fetchUsers = async (query) => {
...
 /**
- * Fetches the user data from the passes userId
+ * Fetches the user data from the the provided username or userId
  *
- * @param userId { string }: User id
+ * @param { Object }: Object with username and userId, any of the two can be used
  * @return {Promise<{userExists: boolean, user: <userModel>}|{userExists: boolean, user: <userModel>}>}
  */
-const fetchUser = async (userId) => {
+const fetchUser = async ({ userId, username }) => {
```
Comment (whyDontI) [inline models/users.js:87]: Add default values,
```
{ userId = null, userName = null }
```

File: models/users.js [hunk 3]
```diff
@@ -79,20 +79,31 @@ const fetchUsers = async (query) => {
...
-    const user = await userModel.doc(userId).get()
+    let userData, id
+    if (username) {
+      const user = await userModel.where('username', '==', username).limit(1).get()
 
+      user.forEach(doc => {
+        id = doc.id
+        userData = doc.data()
+      })
+    } else {
```
Comment (whyDontI) [inline models/users.js:97]: Put a check for userId, both values can also be null.

**Review by whyDontI — CHANGES_REQUESTED** [review]

File: middlewares/authenticate.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ module.exports = async (req, res, next) => {
         })
 
         // add user data to `req.userData` for further use
-        req.userData = await users.fetchUser(decoded.userId)
+        req.userData = await users.fetchUser({ userId: decoded.userId })
```
Comment (whyDontI) [inline middlewares/authenticate.js:61]: Destructure `decoded` it's being used at multiple places

**Review by whyDontI — CHANGES_REQUESTED** [review]

File: models/users.js (hunk 3 above)
Comment (swarajpure) [reply models/users.js:97]: I didn't understand how can both values be null?

File: middlewares/authenticate.js (hunk 4 above)
Comment (swarajpure) [reply middlewares/authenticate.js:61]: Yes sure

File: models/users.js (hunk 2 above)
Comment (swarajpure) [reply models/users.js]: Done.

File: middlewares/authenticate.js (hunk 4 above)
Comment (swarajpure) [reply middlewares/authenticate.js]: Done with requested changes. Please review.

File: models/users.js (hunk 3 above)
Comment (whyDontI) [reply models/users.js]: Edge case

File: middlewares/authenticate.js (hunk 4 above)
Comment (whyDontI) [reply middlewares/authenticate.js]: Did you push your changes?
Can't see destructured `decoded` varialble

File: middlewares/authenticate.js [hunk 5]
```diff
@@ -31,10 +31,10 @@ module.exports = async (req, res, next) => {
       token = req.headers.authorization.split(' ')[1]
     }
 
-    const decoded = authService.verifyAuthToken(token)
+    const { userId } = authService.verifyAuthToken(token)
```
Comment (swarajpure) [inline middlewares/authenticate.js:34]: @whyDontI destructured it here

**Review by whyDontI — APPROVED** [review]
Great work @swarajpure 👍🏽

**Pull request merged**